Mobile-Workshop/Mobile-Workshop/ContentView.swift

```diff
@@ -7,15 +7,113 @@
 
 import SwiftUI
 
-struct ContentView: View {
+struct LoginView: View {
+    @State private var path: [String] = []
+    @State private var email: String = ""
+    @State private var password: String = ""
+    @State private var isEmailValid = true
+    @State private var isPasswordVisible = false
+    @State private var isLoginValid = false
+    
     var body: some View {
         VStack {
-            Image(systemName: "globe")
-                .imageScale(.large)
-                .foregroundColor(.accentColor)
-            Text("Hello, world!")
+            Image("logo")
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(width: 200, height: 200)
+                .padding(.bottom, 40)
+            
+            TextField("Email", text: $email, onEditingChanged: { isEditing in
```

Comment:
Seems like this constructor is [deprecated](https://developer.apple.com/documentation/swiftui/textfield/init(_:text:oneditingchanged:oncommit:)-588cl). If you got time do refactor it, but if not -- make sure to do that later and to not use deprecated ones.
Curious, did the IDE not tell you that by chance (perhaps not updated to latest Xcode version)? 
P.S. You can also use [`onChangeOf`](https://developer.apple.com/documentation/swiftui/view/onchange(of:perform:)) as it is not related to a particular SwiftUI element in combination with a debounce method. We'll be talking about it later for sure, but you can research it yourself if you got the time.